samza-core/src/main/java/org/apache/samza/config/loaders/PropertiesConfigLoader.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.config.loaders;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import org.apache.samza.SamzaException;
+import org.apache.samza.config.Config;
+import org.apache.samza.config.ConfigLoader;
+import org.apache.samza.config.MapConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.util.Objects.requireNonNull;
+
+
+public class PropertiesConfigLoader implements ConfigLoader {
+  private static final Logger LOG = LoggerFactory.getLogger(PropertiesConfigLoader.class);
+
+  private final String path;
+
+  public PropertiesConfigLoader(String path) {
```

Comment:
Instead of using `String` for the type, consider using something like `URI`. It provides a bit more structure/validation. The existing `ConfigFactory` uses `URI`, and it seems like that is something useful to keep.

Reply:
Good question, I am putting String here as we need a relative path on cluster based job coordinator to read configs, URI with file schema does not seem to have a clean support of relative path.

Reply:
Is it not possible to use absolute path? The existing `PropertiesConfigFactory` uses `URI`, so it can access a config file from anywhere. It seems like your `PropertiesConfigLoader` should be consistent with that.
For example, https://github.com/apache/samza-hello-samza uses absolute path.

Reply:
PropertiesConfigFactory is able to use absolute path because it is running during start up and we can specify the absolute path when launching the job.

PropertiesConfigLoader, on the other hand, is being invoked on AM, where the config file is located under the deployment folder, which means, the absolute path contains the deployment folder name, such as "application_1576531941601_0894", which we don't know before submitting the job. So, during job submission, we only specify the relative path to tell where AM should be looking for the config file.

Reply:
Ah, right, the configs will be loaded on the job coordinator. Thanks for clarifying.
Can you please add some more documentation about how to properly specify the path? It's unclear from the code what "path" means and what the format is, since it is just a String.

Reply:
Added more instruction using Hello Samza as an example.